### ISSUE TITLE
[Combobox] List of options collapses when it receives screen reader's focus

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3654]
### 😯 Current Behavior

When user have screen reader turned on and try to expand listbox using arrows focus is placed on the firs option for short moment and then listbox is collapsing. 

Because of that screen reader users can't easily use combobox.

https://github.com/user-attachments/assets/d0151c77-ee26-4181-8984-24274b5c40b5

This happens for VO in Chrome and Safari and for NVDA in Chrome and Edge. Does not happen on Firefox on both Win11 and macOs.

### 🤔 Expected Behavior/possible solution?

When screen reader user place focus inside combobox input and press `arrow down` listbox should be expanded and focus should be placed on the first option inside listbox. Focus should remain inside listbox until user close listbox with `esc` key or choose option in single select combobox.

### 🖥️ Steps to Reproduce

1. With screen reader turned on go to https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--basic
2. Place focus inside combobox
3. Press arrow down
4. Verify if listbox remains visible and if arrows down and up can be used to move between options

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--basic

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Safari, Chrome

### Component version

0.1.11

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
